src/test/java/com/commercetools/paymenttoorderprocessor/ReadMessagesJobIntegrationTest.java

```diff
@@ -0,0 +1,163 @@
+import io.sphere.sdk.carts.commands.CartDeleteCommand;
+import io.sphere.sdk.carts.commands.CartUpdateCommand;
+import io.sphere.sdk.carts.commands.updateactions.AddPayment;
+import io.sphere.sdk.client.BlockingSphereClient;
+import io.sphere.sdk.http.HttpStatusCode;
+import io.sphere.sdk.messages.Message;
+import io.sphere.sdk.messages.queries.MessageQuery;
+import io.sphere.sdk.models.Address;
+import io.sphere.sdk.models.LocalizedString;
+import io.sphere.sdk.payments.*;
+import io.sphere.sdk.payments.commands.PaymentUpdateCommand;
+import io.sphere.sdk.payments.commands.updateactions.AddTransaction;
+import io.sphere.sdk.payments.commands.updateactions.ChangeTransactionState;
+import io.sphere.sdk.projects.Project;
+import io.sphere.sdk.projects.queries.ProjectGet;
+import io.sphere.sdk.queries.PagedQueryResult;
+import io.sphere.sdk.taxcategories.TaxCategory;
+import io.sphere.sdk.taxcategories.queries.TaxCategoryQuery;
+import org.assertj.core.api.Fail;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.batch.core.Job;
+import org.springframework.batch.core.JobParameters;
+import org.springframework.batch.core.launch.JobLauncher;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.ConfigFileApplicationContextInitializer;
+import org.springframework.boot.test.context.SpringBootContextLoader;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static com.commercetools.paymenttoorderprocessor.fixtures.PaymentFixtures.EUR;
+import static com.commercetools.paymenttoorderprocessor.fixtures.PaymentFixtures.EUR_20;
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = {ReadMessagesJob.class, BasicTestConfiguration.class, ExtendedTestConfiguration.class,
+        ShereClientConfiguration.class, MessageReaderIntegrationTest.ContextConfiguration.class,
+        HttpClientMockConfiguration.class},
+        initializers = ConfigFileApplicationContextInitializer.class,
+        loader = SpringBootContextLoader.class)
+public class ReadMessagesJobIntegrationTest extends IntegrationTest {
+
+    @Autowired
+    private BlockingSphereClient testClient;
+
+    @Autowired
+    private Job job;
+
+    @Autowired
+    private JobLauncher jobLauncher;
+
+    @Autowired
+    private HttpClientMock httpClientMock;
+
+    @Autowired
+    private MessageProcessedManager messageProcessedManager;
+
+    @Value("${ctp.custom.object.containername}")
+    private String customObjectContainerName;
+
+    @Test
+    public void readAndProcessMessagesJobIntegrationTest() throws Exception {
```

Comment:
this test returns with output, I am not sure for the rests, it might be I don't have a valid endpoint for the order creation part.

{"message":"Request is successful with status 200, but order is not created. Cart [47175e08-b6c1-4c35-9e5d-8519ce6045b0] is not processed. Reason:  ","timestamp":"2021-06-10T12:38:40.442+02:00","loggerName":"com.commercetools.paymenttoorderprocessor.jobs.actions.OrderCreator","severity":"INFO"}

Reply:
The order endpoint is mocked using this class: https://github.com/commercetools/commercetools-payment-to-order-processor/blob/master/src/test/java/com/commercetools/paymenttoorderprocessor/testconfiguration/HttpClientMock.java. That's why we get the log message.